tmux: reorder tmux.conf -> put `extraConfig` in the end

jorsn opened the pull request:
This allows overriding of every tmux.conf setting via `extraConfig`, even plugin settings.

See also [#875](https://github.com/rycee/home-manager/pull/875#issuecomment-545555727).

jorsn commented on modules/programs/tmux.nix:
```diff
@@ -30,7 +30,16 @@ let
+  '';
+
   tmuxConf = ''
+    ${optionalString cfg.sensibleOnTop tmuxConfSensible}
```
I could also directly inline this. Doesn't really matter to me.

rycee commented on modules/programs/tmux.nix:
```diff
@@ -263,58 +300,18 @@ in
-        '';
-      })
+      # config file ~/.tmux.conf
+      # use inverse order, because mkMerge inverts it
```
Better use `mkBefore` and `mkAfter` to control order since `mkMerge` does not make any guarantee w.r.t ordering.

jorsn replied:
In the update just pushed, I use explicit ordering.

rycee commented:
@kalbasit Do you have time to have a quick look at this? I don't really know anything about tmux 🙂

jorsn commented:
I just pushed a new version where I
* split the commit into the real change and the refactoring
* rebased it  this onto #1195. This solves the merge conflict with master which had arisen.

### Checklist

*(does not cover 1fd475c, bcf8aff)*

- [x] Code formatted with `./format`.

- [ ] Code tested through `nix-shell --pure tests -A run.all`.

- [x] Code tested through tmux related tests:
  ```
  nix-shell --pure tests -A list \
  | sed -nE 's/(.*tmux.*)/run.&/p' \
  | xargs -i nix-shell --pure tests -A '{}'
  ```

- [x] Test cases updated, not added.

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

jorsn commented:
- rebased onto master after #1195 was accepted
- edited last note accordingly

rycee commented:
Thanks! Rebased to master in a7cdfaa32585fce404cf8890bae099348e53e0ad.
